models/DataObject/ClassDefinition/Data/User.php

```diff
@@ -94,6 +104,23 @@ public function getDataForResource($data, $object = null, $params = [])
     public function configureOptions()
     {
         $list = new Model\User\Listing();
+
+        if (!empty($this->restrictTo)) {
+            $roles = explode(',', $this->restrictTo);
+            $expr = \Pimcore\Db::getConnection()->createQueryBuilder()->expr();
+
+            foreach ($roles as $roleId) {
+                $list->setCondition(
+                    $expr->orX(
```

Comment:
I think it would make sense to store this in a repository, but where is the best place to do this in the Pimcore context?

So `User\Listing::getByRoleId()` or `Model\User\Role::getById(1)->getUsers()` or something like that, what do you think?

Reply:
But I think this is a bigger todo for a future issue / PR to improve the roles query between user and role